driver/src/db_interface.rs

```diff
@@ -116,24 +115,24 @@ impl DbInterface for MongoDB {
     fn get_standing_orders_of_slot(
         &self,
         slot: u32,
-    ) -> Result<Vec<models::StandingOrder>, DriverError> {
+    ) -> Result<[models::StandingOrder; models::NUM_RESERVED_ACCOUNTS], DriverError> {
         let pipeline = vec![
             doc!{"$match" => (doc!{"validFromAuctionId" => (doc!{ "$lte" => slot})})},
             doc!{"$sort" => (doc!{"validFromAuctionId" => -1, "_id" => -1})},
             doc!{"$group" => (doc!{"_id" => "$accountId", "orders" => (doc!{"$first" =>"$orders" }), "batchIndex" => (doc!{"$first" => "$batchIndex" })})}
         ];
 
-        info!("Querying standing_orders: {:?}", pipeline);
+               info!("Querying standing_orders: {:?}", pipeline);
         let non_zero_standing_orders = self.client
             .db(models::DB_NAME)
             .collection("standing_orders")
             .aggregate(pipeline, None)?
             .map(|d| d.map(models::StandingOrder::from).map_err(DriverError::from))
             .collect::<Result<Vec<_>, _>>()?;
 
-        let mut standing_orders = (0..models::NUM_RESERVED_ACCOUNTS)
-            .map(|k| models::StandingOrder::empty(u16::from(k)))
-            .collect::<Vec<models::StandingOrder>>();
+        let mut standing_orders = models::StandingOrder::empty(0).initialize_array();
+        (0..models::NUM_RESERVED_ACCOUNTS)
```

Comment:
I am not sure I understand this line here... Does it connect to the previous line? If so, then this appears to be an indentation issue, but if not what is it?

Reply:
It just creates an iterator for the numbers 0 to NUM_RESEVER_ACCOUNTS. Yeah it looks a little bit unsual